### ISSUE TITLE
Suggstion: Copy other open source messaging app code?

### ISSUE THREAD

**Issue by trymeouteh** [Tox/Tox-Client-Standard#38]
Tox looks very promising due to it not requiring a phone number and being decentralised. But it lacks many features compared to other open source messaging apps such as Signal, Riot, Wire, Jami and so on.

I am not a programmer but could you copy and modify some of the code from these other projects and add these features into Tox to help its develop and to get Tox developed quicker than building it from the ground up?

**Comment by Zer0-One** [comment]
Are you just pretending

**Comment by JFreegman** [comment]
I agree. Since the code is open source we should just be able to copy paste most of it.

**Comment by notgne2** [comment]
agreed, by copy pasting the source code of signal I was able to get all signal's features, so presumably if you copy paste the code of signal into tox you should get both

**Comment by ghost** [comment]
If it was that easy, it would have been done. Unfortunately you need to be really careful when copying—you don't want to copy all the bugs and misfeatures!
